src/sagemaker/experiments/run.py

```diff
@@ -192,7 +195,31 @@ def __init__(
                 self.run_name,
                 self.experiment_name,
             )
-        self._trial.add_trial_component(self._trial_component)
+
+        def search_trial_component_associated_trial():
+            search_results = sagemaker_session.sagemaker_client.search(
+                Resource="ExperimentTrialComponent",
+                SearchExpression={
+                    "Filters": [
+                        {
+                            "Name": "TrialComponentName",
+                            "Operator": "Equals",
+                            "Value": str(self._trial_component.trial_component_name),
+                        },
+                        {
+                            "Name": "Parents.TrialName",
+                            "Operator": "Equals",
+                            "Value": str(self._trial.trial_name),
+                        },
+                    ]
+                },
+            )
+            if search_results["Results"]:
+                return True
+            return False
```

Comment:
* while this works syntactically its not idea style 
* see https://google.github.io/styleguide/pyguide.html for general style guide
* for now move this to a private method i.e. `_search_trial_component_associated_trial`
* consider renaming the method to be clear it returns a boolean i.e. `_trial_component_is_associated_to_trial`

Reply:
also this method might be more appropriate on the tc class i.e. `self._trial_component.is_associated_to_trial(...)`